app/src/main/java/rick_and_morty/ui/episodes/EpisodesViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package rick_and_morty.ui.episodes
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import rick_and_morty.data.repository.EpisodesRepository
+import javax.inject.Inject
+
+@HiltViewModel
+class EpisodesViewModel @Inject constructor(
+    private val episodesRepository: EpisodesRepository,
+) : ViewModel() {
+
+    private var page = 1
+
+    private val _episodes = MutableStateFlow(EpisodesUiState())
+    val episodes: StateFlow<EpisodesUiState> = _episodes
+
+    init {
+        getEpisodes()
+    }
+
+    fun getEpisodes() {
+        _episodes.update { it.copy(isLoading = true) }
+        viewModelScope.launch {
+            try {
+                val getEpisodes = episodesRepository.getEpisodes(page)
+                _episodes.update {
+                    it.copy(
+                        isLoading = false,
+                        episodeResults = if (getEpisodes != null)
+                            it.episodeResults + getEpisodes
+                        else it.episodeResults
+                    )
+                }
+                page += 1
```

Comment:
I think it would be good to check if `getEpispdes` is `null` before incrementing `page`. 
Otherwise, in case we would not get data for page 2 because of a network issue, we would then request data for page 3 in the next try.

Reply:
Yep, I have noticed the issue here, it is also same for the CharacterViewModel. I thought to make another PR for this and for the Tests which left

Reply:
Why another PR?
put it in this one.
Especially the tests. They are a MUST in every PR.

Reply:
Ok

Reply:
Done